main.go

```diff
@@ -132,7 +134,24 @@ func tr1d1um(arguments []string) (exitCode int) {
 
 	APIRouter := r.PathPrefix(fmt.Sprintf("/%s/", apiBase)).Subrouter()
 
-	authenticate, err = authenticationHandler(v, logger, metricsRegistry)
+	u := v.Sub(traceProviderKey)
+	if u == nil {
+		fmt.Fprintf(os.Stderr, "traceProvider configuration is missing.\n")
+		return 1
+	}
+	config := &candlelight.Config{
+		ApplicationName: applicationName,
+	}
+	headerConfig := &candlelight.HeaderConfig{}
+	u.Unmarshal(headerConfig)
+	u.Unmarshal(config)
+	traceProvider, err := candlelight.ConfigureTracerProvider(*config)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Unable to build traceProvider: %s\n", err.Error())
+		return 1
+	}
+	traceConfig := candlelight.TraceConfig{*headerConfig, traceProvider}
```

Comment:
It might be worth creating a helper config struct here. Something like:
```
type tracingConfig struct{
    provider candlelight.Config
    headers candlelight.HeaderConfig
}
```

which maps nicely to a section in the configuration file:
```
---
    tracing:
      provider:
        applicationName: "tr1d1um"
      headers:
        SpanIDHeaderName: "span-id"
```
so that you could just unmarshal straight to the struct and use the values from there.

Reply:
We should move tracingConfig struct to candlelight since it will be used across microservices.

Reply:
Yeah, I think that'd be a good idea!

Reply:
@joe94  Here is the PR for the same https://github.com/xmidt-org/candlelight/pull/19.